packages/angular/common/src/directives/navigation/router-outlet.ts

```diff
@@ -42,7 +42,9 @@ import { RouteView, StackDidChangeEvent, StackWillChangeEvent, getUrl, isTabSwit
   inputs: ['animated', 'animation', 'mode', 'swipeGesture'],
 })
 // eslint-disable-next-line @angular-eslint/directive-class-suffix
-export class IonRouterOutlet implements OnDestroy, OnInit {
+export abstract class IonRouterOutlet implements OnDestroy, OnInit {
+  abstract outletContent: any;
```

Comment:
Previously we were passing in the outletContent via the constructor. However, newer versions of Angular warn that this will override the base implementation. However, this is what we want so I made the base class an abstract class to indicate that extending classes need to implement outletContent.